### PR TITLE
8263899: HttpClient throws NPE in AuthenticationFilter when parsing www-authenticate head

### PR DESCRIPTION
**Pull request by Michael-Mc-Mahon**
Could I get the following simple fix reviewed please? 

The bug is jdk.internal.net.http.AuthenticationFilter throws an NPE wrapped in an IOException if it receives a HTTP response containing a WWW-Authenticate header with no value.

Thanks,
Michael.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8263899](https://bugs.openjdk.java.net/browse/JDK-8263899): HttpClient throws NPE in AuthenticationFilter when parsing www-authenticate head

### Reviewers
 * [Daniel Fuchs](https://openjdk.java.net/census#dfuchs) (@dfuch - **Reviewer**)
 * [Chris Hegarty](https://openjdk.java.net/census#chegar) (@ChrisHegarty - **Reviewer**)

### Download
To checkout this PR locally:
`$ git fetch https://git.openjdk.java.net/jdk pull/3113/head:pull/3113`
`$ git checkout pull/3113`

To update a local copy of the PR:
`$ git checkout pull/3113`
`$ git pull https://git.openjdk.java.net/jdk pull/3113/head`

### CODE REVIEW

File: test/jdk/java/net/httpclient/EmptyAuthenticate.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+    public static void main(String[] args) throws IOException, URISyntaxException, InterruptedException {
+        int port = 8000;
+
+        //start server:
+        HttpServer server = HttpServer.create(new InetSocketAddress(port), 0);
+        server.createContext("/", exchange -> {
+            String response = "test body";
+            exchange.getResponseHeaders().add("www-authenticate", ""); //this empty header will make the HttpClient crash
+            exchange.sendResponseHeaders(401, response.length());
```
Comment (ChrisHegarty) [inline test/jdk/java/net/httpclient/EmptyAuthenticate.java:51]: s/crash/throw NPE

File: test/jdk/java/net/httpclient/EmptyAuthenticate.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+
+public class EmptyAuthenticate {
+
+    public static void main(String[] args) throws IOException, URISyntaxException, InterruptedException {
+        int port = 8000;
```
Comment (ChrisHegarty) [inline test/jdk/java/net/httpclient/EmptyAuthenticate.java:44]: I assume that this hardcoded port no. can be removed?

File: test/jdk/java/net/httpclient/EmptyAuthenticate.java (hunk 1 above)
Comment (Michael-Mc-Mahon) [reply test/jdk/java/net/httpclient/EmptyAuthenticate.java:52]: Will do

File: test/jdk/java/net/httpclient/EmptyAuthenticate.java (hunk 2 above)
Comment (Michael-Mc-Mahon) [reply test/jdk/java/net/httpclient/EmptyAuthenticate.java]: Yes, had just noticed that and pushed a change.

File: test/jdk/java/net/httpclient/EmptyAuthenticate.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+            OutputStream os = exchange.getResponseBody();
+            os.write(response.getBytes());
+            os.close();
+        });
+        server.start();
+
+        HttpResponse<String> response = null;
+        //run client:
+        try {
+            HttpClient httpClient = HttpClient.newHttpClient();
```
[the remark this answers is not in the document]
Comment (Michael-Mc-Mahon) [reply test/jdk/java/net/httpclient/EmptyAuthenticate.java:63]: Yes, the test does not set an authenticator and will fail without the fix.

**Review by dfuch — APPROVED** [review]

File: test/jdk/java/net/httpclient/EmptyAuthenticate.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (dfuch) [reply test/jdk/java/net/httpclient/EmptyAuthenticate.java:63]: LGTM

**Review by ChrisHegarty — APPROVED** [review]

**Pull request closed without merging**